### PR TITLE
fix: set flag key on `$flagd.flagKey`

### PR DESCRIPTION
**Pull request by toddbaert**
As per flagd spec, flag key should be injected into the `$flagd` props bucket.

- moved flag key to correct location
- updated associated code
- added tests asserting location

All existing tests (including fractional integration tests) pass.

### CODE REVIEW

File: providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/OperatorTest.java [hunk 1]
```diff
@@ -18,6 +19,20 @@ static void setUp() {
         OPERATOR = new Operator();
     }
 
+    @Test
+    void flagKeyPresent() throws TargetingRuleException {
```
Comment (toddbaert) [inline providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/OperatorTest.java:23]: new test asserts that `$flagd.flagKey` === the flag key

File: providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/Operator.java [hunk 2]
```diff
@@ -53,7 +59,10 @@ static class FlagProperties {
             if (from instanceof Map) {
                 Map<?, ?> dataMap = (Map<?, ?>) from;
 
-                Object flagKey = dataMap.get(FLAG_KEY);
+                Object flagKey = Optional.ofNullable(dataMap.get(FLAGD_PROPS_KEY))
+                    .filter(flagdProps -> flagdProps instanceof Map)
+                    .map(flagdProps -> ((Map<?, ?>)flagdProps).get(FLAG_KEY))
+                    .orElse(null);
```
Comment (toddbaert) [inline providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/Operator.java:65]: @Kavindu-Dodan there's not a lot of Optional usage in this repo, but the alternative was really ugly.

File: providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/Operator.java [hunk 3]
```diff
@@ -53,15 +59,18 @@ static class FlagProperties {
             if (from instanceof Map) {
                 Map<?, ?> dataMap = (Map<?, ?>) from;
 
-                Object flagKey = dataMap.get(FLAG_KEY);
+                final Object flagKey = Optional.ofNullable(dataMap.get(FLAGD_PROPS_KEY))
```
Comment (toddbaert) [inline providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/Operator.java:62]: @Kavindu-Dodan there's not a lot of Optional usage in this repo, but the alternative was really ugly and not immutable.

Comment (Kavindu-Dodan) [reply providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/Operator.java:62]: This looks fine :)

File: providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/OperatorTest.java (hunk 1 above)
Comment (Kavindu-Dodan) [reply providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/OperatorTest.java:23]: Nice !  this is a  really cool test 👍

**Review by Kavindu-Dodan — APPROVED** [review]
LGTM & thanks for fixing this

**Pull request merged**